trio/_core/_ki.py

```diff
@@ -170,26 +171,31 @@ def wrapper(*args, **kwargs):
 disable_ki_protection.__name__ = "disable_ki_protection"
 
 
-@contextmanager
-def ki_manager(deliver_cb, restrict_keyboard_interrupt_to_checkpoints):
-    if (
-        not is_main_thread()
-        or signal.getsignal(signal.SIGINT) != signal.default_int_handler
-    ):
-        yield
-        return
-
-    def handler(signum, frame):
-        assert signum == signal.SIGINT
-        protection_enabled = ki_protection_enabled(frame)
-        if protection_enabled or restrict_keyboard_interrupt_to_checkpoints:
-            deliver_cb()
-        else:
-            raise KeyboardInterrupt
-
-    signal.signal(signal.SIGINT, handler)
-    try:
-        yield
-    finally:
-        if signal.getsignal(signal.SIGINT) is handler:
-            signal.signal(signal.SIGINT, signal.default_int_handler)
+@attr.s
+class KIManager:
```

Comment:
Merge with master so this doesn't show up as part of the guest mode PR?

Reply:
I did merge with master, and diff you're pointing at actually is new in the guest mode PR :-). Master has the change to flip the default control-C protection status; this change is to convert `ki_manager` from a context manager (which is awkward to wrap around guest mode) into a regular object with `install`/`close` methods.

Reply:
I'm not real happy with how the `KIManager` stuff is structured, so if you have any ideas on how to do it better than I'm interested, and it's probably worth scrutinizing a bit in review b/c everything around KI is tricky. But this does seem workable to me and we can always improve it later.

Reply:
Oh, whoops! I was going "I've seen this change before", but in fact I saw it in a previous commit on this diff. I'm happy with this approach.